packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1892,7 +1923,15 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
       final Rect localRect = Rect.fromLTWH(0, 0, selectables[index].size.width, selectables[index].size.height);
       final Matrix4 transform = selectables[index].getTransformTo(null);
       final Rect globalRect = MatrixUtils.transformRect(transform, localRect);
-      if (globalRect.contains(event.globalPosition)) {
+      final List<Rect> rectsInside = _selectableContains(selectables[index], selectables);
```

Comment:
Is this making this O(n^2) instead of O(n) before?  Not sure if this is a bottle neck at all, just pointing it out.

Reply:
Yeah this would be O(n^2) now. I couldn't think of a re-factor for the current solution that would make this O(n). @chunhtai do you have any thoughts on this?

Reply:
I am not sure if this is the right fix. I though we may run into the same problem even if the inner selectable is not completely contained by the outer one?

In the example 
![image](https://github.com/flutter/flutter/assets/47866232/dd5df403-2862-4110-b91b-6a8d3d8c3f89)
It seems to me the right behavior is that if the selectword at the inside of inner selectable, the outer box should return SelectionResult.next because it should have known the coordinate is outside its actual text layout.

Reply:
That makes a lot of sense! That makes the solution a lot simpler and keeps this O(n). Thanks for the input!